src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/TextFieldIndexProvider.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentFields.Fields;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata;
+using YesSql.Indexes;
+
+namespace OrchardCore.ContentFields.Indexing
+{
+    public class TextFieldIndex : ContentFieldIndex
+    {
+        public string Text { get; set; }
+        public string BigText { get; set; }
+    }
+
+    public class TextFieldIndexProvider : ContentFieldIndexProvider
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly HashSet<string> _ignoredTypes = new HashSet<string>();
+        private IContentDefinitionManager _contentDefinitionManager;
+
+        public TextFieldIndexProvider(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public override void Describe(DescribeContext<ContentItem> context)
+        {
+            context.For<TextFieldIndex>()
+                .Map(contentItem =>
+                {
+                    // Can we safely ignore this content item?
+                    if (_ignoredTypes.Contains(contentItem.ContentType))
+                    {
+                        return null;
+                    }
+
+                    if (!contentItem.Latest && !contentItem.Published)
+                    {
+                        return null;
+                    }
+
+                    // Lazy initialization because of ISession cyclic dependency
+                    _contentDefinitionManager = _contentDefinitionManager ?? _serviceProvider.GetRequiredService<IContentDefinitionManager>();
+
+                    // Search for Text fields
+                    var fieldDefinitions = _contentDefinitionManager
+                        .GetTypeDefinition(contentItem.ContentType)
+                        .Parts.SelectMany(x => x.PartDefinition.Fields.Where(f => f.FieldDefinition.Name == nameof(TextField)))
+                        .ToArray();
+
+                    var results = new List<TextFieldIndex>();
+
+                    foreach (var fieldDefinition in fieldDefinitions)
+                    {
+                        var jPart = (JObject)contentItem.Content[fieldDefinition.PartDefinition.Name];
+
+                        if (jPart == null)
+                        {
+                            continue;
+                        }
+
+                        var jField = (JObject)jPart[fieldDefinition.Name];
+
+                        if (jField == null)
+                        {
+                            continue;
+                        }
+
+                        var field = jField.ToObject<TextField>();
+
+                        if (!String.IsNullOrEmpty(field.Text))
```

Comment:
Why? Wouldn't we want to search for an empty string?

Reply:
You are right